### PR TITLE
Remove the need to use a singleton to get the ECM pointer

### LINKED ISSUE #378: Simplify logic to get the Entity Component Manager and Event Manager
In Ignition Gazebo, the [`Server`](https://github.com/ignitionrobotics/ign-gazebo/blob/24be610ac1/include/ignition/gazebo/Server.hh) class contains one or multiple [`SimulationRunner`](https://github.com/ignitionrobotics/ign-gazebo/blob/24be610ac1/src/SimulationRunner.hh) objects (in most cases, just one), each simulating an independent world. The data storage of each `SimulationRunner` is implemented in a Entity Component System.

Historically, the ECS has always been hidden from the user (https://github.com/ignitionrobotics/ign-gazebo/issues/248). Though, the entire Gazebo ScenarIO logic is based on the fact that we can access and alter this data storage. Until now, we had to rely on a kind of dirty workaround: a plugin that implements only the [`ECMProvider::Configure`](https://github.com/robotology/gym-ignition/blob/311b08ef16/cpp/scenario/plugins/ECMProvider/ECMProvider.cpp) callback that gets the pointer to the ECM (and the event manager) and stores it in a singleton. Then, within the same process, the singleton could be used to get these pointers and consume them.

It's easy to see that the combination of plugins + singletons + python could create problems. In fact, we struggled with some hairy issues in the past (see a recap in #354 that, even if not specific to this, is highly related).

Luckily, the new upstream https://github.com/ignitionrobotics/ign-gazebo/pull/936 allows us to gather the ECM without the need of using the plugin + singleton :tada: If I understood, the `ECMProvider` class of the plugin can be allocated together with the `Server` (e.g. in the `GazeboSimulator` class) and its system directly added. Then, the ECM and event manager can be gathered from the `ECMProvider` object with new getters, and stored, if needed, in our `GazeboSimulator` class.

Likely this solves for good the need to alter the dlopen flags as described in #7 (very old issue!). Though, if it does not harm anyone, I'd leave this workaround there in any case.

### PR DESCRIPTION
**Pull request by diegoferigo**
Closes #378 (thanks to https://github.com/ignitionrobotics/ign-gazebo/pull/936, further details in https://github.com/ignitionrobotics/ign-gazebo/issues/248) and implements #289.

Requires https://github.com/ignitionrobotics/ign-gazebo/pull/953. Merged now in upstream, waiting forward porting to Edifice.

### CODE REVIEW

File: scenario/src/gazebo/src/GazeboSimulator.cpp
Comment (diegoferigo) [inline scenario/src/gazebo/src/GazeboSimulator.cpp:735]: Reminder: address this before merging

**Review by traversaro — APPROVED** [review]
Approving, even if I guess you want to wait for a release of the required changes to merge this.

**Comment by diegoferigo** [comment]
> Approving, even if I guess you want to wait for a release of the required changes to merge this.

Yes, I need https://github.com/ignitionrobotics/ign-gazebo/pull/953 to be forward ported to upstream's ign-gazebo5 (Edifice) in order to merge the PR to `devel`, and a new Ignition Gazebo minor release to merge `devel` in `master`.

**Comment by diegoferigo** [comment]
Forward porting has been done in https://github.com/ignitionrobotics/ign-gazebo/pull/968. CI is happy, at least to target `devel`, merging.

**Pull request merged**
